File: 1308/1308.cpp
class Solution {
public:
    int runningTotalForDifferentGender(TreeNode* root) {
        if(!root) return 0;
        return 1+max(runningTotalForDifferentGender(root->left),runningTotalForDifferentGender(root->right));
    }
};

File: 1308/1308.cs
public class Solution {
    public int RunningTotalForDifferentGender(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(RunningTotalForDifferentGender(root.left), RunningTotalForDifferentGender(root.right));
    }
}